addons/mail/static/src/components/thread_view/thread_view.xml

```diff
@@ -23,6 +23,16 @@
                         t-ref="messageList"
                     />
                 </t>
+                <t t-elif="threadView.threadCache.hasLoadingFailed">
```

Comment:
Shouldn't we display the error in priority? If the last fetch failed, it might be better to let the user know it failed, instead of showing obsolete messages if those have been loaded fine previously?

The current order might lead to the user having obsolete information displayed and no way to know about it.

Maybe there should be a new case inside "isLoaded" case, so that a banner is added on top or bottom of the thread if both "isLoaded" and "hasLoadingFailed" are true at the same time, to say it wasn't possible to refresh the thread, click here to retry?

Reply:
> Shouldn't we display the error in priority? If the last fetch failed, it might be better to let the user know it failed, instead of showing obsolete messages if those have been loaded fine previously?

I don't understand how this could happen ? As soon there is an error, `isLoaded` is false so no message will be shown.

> Maybe there should be a new case inside "isLoaded" case, so that a banner is added on top or bottom of the thread if both "isLoaded" and "hasLoadingFailed" are true at the same time, to say it wasn't possible to refresh the thread, click here to retry?

This I think could happen, if there is enough messages on the thread and the first bunch of messages works but the next bunch fails. Actually all previous messages will be removed and the error will be shown.

Reply:
Maybe ask AJU about this case ?

Reply:
I don't think `isLoaded` is ever reset to false as soon as it has been `true` once.

Reply:
Mmh indeed... should it ? in a sense, if the cache is marked as "isLoaded" but a new loading is triggered and this one failed ... so the cache is no more loaded, no ?

Reply:
The current behavior of `isLoaded` is fine: as soon as there are messages to display, it's good to display them, even if they are obsolete. Likewise, if we have messages loaded, and we are "loading" again, it is fine to keep displaying the existing messages while it is loading (more than being fine, it is also important to prevent flicker and annoyances).

I just thing we should have better indicators that thread is loading/failed when it is already loaded and messages are displayed.

Reply:
so basically :
- if no messages and thread load failed : we keep the current behaviour which is showing an error message on the whole thread view with a retry button
- if messages and thread load failed : we display messages and we add an alert in top of the thread (should it depend on ASC/DESC order of the message list ?) saying there was an issue with the load and a retry button (basically I think it can only happen with loadMore and loadNew messages)

is that what is in your mind ?

Reply:
Yes, and also it can happen with non-main thread cache which have to be reloaded when a new message is received on the corresponding thread (eg. if we are filtering on a keyword in discuss, receiving a new message on thread will lead to a reload of the filtered view to include the newly received message, only if it matches the search).

Probably should depend on asc/desc indeed, or always at the top. The only case I would avoid for sure is to display it at the bottom in chatter, that makes no sense. But for discuss & chat window, having it at the top or at the bottom would seem fine either way (similar to out of office & visitor info which are always at the top).